src/descriptor_runner/descriptor_runner/descriptor_runner_fallback.ts

```diff
@@ -40,12 +40,12 @@ export default class DescriptorRunnerFallback extends DescriptorRunner<GraphDesc
         //nothing to do
     }
 
-    async load(directory: string, progressCallback?: (loaded: number, total: number) => any) {
+    async load(directory: string, progressCallback?: (loaded: number, total: number) => any, weightDirectory?: string) {
         let [descriptor, weightRawArray] = await Promise.all([
             webdnnFetch(`${directory}/graph_${this.backendName}.json`, {ignoreCache: this.ignoreCache})
                 .then(res => res.json() as Promise<GraphDescriptorFallback>),
 
-            webdnnFetch(`${directory}/weight_${this.backendName}.bin`, {ignoreCache: this.ignoreCache})
+            webdnnFetch(`${weightDirectory || directory}/weight_${this.backendName}.bin`, {ignoreCache: this.ignoreCache})
```

Comment:
Please set `weightDirectory` url not here but in `WebDNN.load` by using `registerTransformUrlDelegate` .